packages/neon-core/src/tx/transaction/Transaction.ts

```diff
@@ -1,104 +1,379 @@
+import { ScriptIntent, createScript } from "../../sc";
+const log = logger("tx");
 
-function getType(type: number): any {
-  switch (type) {
-    case 0x00:
-      return MinerTransaction;
-    case 0x01:
-      return IssueTransaction;
-    case 0x02:
-      return ClaimTransaction;
-    case 0x20:
-      return EnrollmentTransaction;
-    case 0x40:
-      return RegisterTransaction;
-    case 0x80:
-      return ContractTransaction;
-    case 0xd0:
-      return PublishTransaction;
-    case 0xd1:
-      return InvocationTransaction;
-    case 0x90:
-      return StateTransaction;
-    default:
-      throw new Error(`Unknown TransactionType: ${type}`);
-  }
+export interface TransactionLike {
+  version: number;
+  nonce: number;
+  sender: string;
+  systemFee: Fixed8 | number;
+  networkFee: Fixed8 | number;
+  validUntilBlock: number;
+  attributes: TransactionAttributeLike[];
+  intents: ScriptIntent[];
+  scripts: WitnessLike[];
+  script: string;
 }
 
-/**
- * @class Transaction
- * @classdesc
- * Transactions are what you use to interact with the blockchain.
- * A transaction is made up of components found in the component file.
- * Besides those components which are found in every transaction, there are also special data that is unique to each transaction type. These 'exclusive' data can be found in the exclusive file.
- * This class is a wrapper around the various transaction building methods found in this folder.
- */
-export class Transaction extends BaseTransaction {
+export class Transaction {
+  public version: number;
+  /**
+   * A random number to avoid hash collision
+   */
+  public nonce: number;
+  /**
+   * transation invoker in script hash
+   */
+  public sender: string;
+
+  /**
+   * Distributed to NEO holders
+   */
+  public systemFee: Fixed8;
+
+  /**
+   * Distributed to consensus nodes
+   */
+  public networkFee: Fixed8;
+
+  /**
+   * System Fee calculated while adding intents
+   */
+  private _pre_systemFee: Fixed8;
+
+  /**
+   * Current transaction will be invalid after block of height validUntilBlock
+   */
+  public validUntilBlock: number;
+  public attributes: TransactionAttribute[];
+  public intents: ScriptIntent[];
+  public scripts: Witness[];
+  public script: string;
+
+  public static MAX_VALIDUNTILBLOCK_INCREMENT = 2102400;
+
+  public constructor(tx: Partial<TransactionLike> = {}) {
+    const {
+      version,
+      nonce,
+      sender,
+      systemFee,
+      networkFee,
+      validUntilBlock,
+      attributes,
+      intents,
+      scripts,
+      script
+    } = tx;
+    this.version = version || TX_VERSION;
+    this.nonce = nonce || parseInt(ab2hexstring(generateRandomArray(4)), 16);
+    this.sender = formatSender(sender);
+    // TODO: The default should be snapshot.height + MAX_VALIDUNTILBLOCK_INCREMENT, but it needs request to get block height, thus this is a temporary value
+    this.validUntilBlock =
+      validUntilBlock || Transaction.MAX_VALIDUNTILBLOCK_INCREMENT;
```

Comment:
default should be a illegal value like 0 / -1 so we can quickly check its validity. Dont try to magic it in the constructor.

Reply:
ok, changed to 0